runtime/cpp/instance_segmentation/yolov5seg/yolov5seg_example.cpp

```diff
@@ -1,404 +1,318 @@
+    uint8_t *src_ptr = static_cast<uint8_t*>(src_buffer.data());
+    uint16_t detections_count = *(uint16_t*)src_ptr;
+    detections.reserve(detections_count);
+    size_t buffer_offset = sizeof(uint16_t);
+    for (size_t i = 0; i < detections_count; i++) {
+        hailo_detection_with_byte_mask_t detection = *(hailo_detection_with_byte_mask_t*)(src_ptr + buffer_offset);
+        buffer_offset += sizeof(hailo_detection_with_byte_mask_t) + detection.mask_size;
+        detections.emplace_back(std::move(detection));
+    }
+    return detections;
 }
 
-template <typename T>
-hailo_status run_inference(std::vector<InputVStream>& input_vstream, std::vector<OutputVStream>& output_vstreams, std::string video_path,
-                    std::chrono::time_point<std::chrono::system_clock>& write_time_vec,
-                    std::vector<std::chrono::time_point<std::chrono::system_clock>>& read_time_vec,
-                    std::chrono::duration<double>& inference_time, std::chrono::duration<double>& postprocess_time, 
-                    double frame_count, double org_height, double org_width) {
-
-    hailo_status status = HAILO_UNINITIALIZED;
-    
-    auto output_vstreams_size = output_vstreams.size();
-
-    std::vector<std::shared_ptr<FeatureData<T>>> features;
-    features.reserve(output_vstreams_size);
-    for (size_t i = 0; i < output_vstreams_size; i++) {
-        std::shared_ptr<FeatureData<T>> feature(nullptr);
-        auto status = create_feature<T>(output_vstreams[i].get_info(), output_vstreams[i].get_frame_size(), feature);
-        if (HAILO_SUCCESS != status) {
-            std::cerr << "Failed creating feature with status = " << status << std::endl;
-            return status;
+template <typename T> cv::Mat semseg_post_process(std::vector<T>& logits, int width, int height, cv::Mat& frame) {
+    std::vector<hailo_detection_with_byte_mask_t> detections = convert_nms_with_byte_mask_buffer_to_detections(logits);
+    cv::Mat overlay = cv::Mat::zeros(height, width, CV_8UC3);
+
+    // if (frame.channels() == 3){
+    //     cv::cvtColor(frame, frame, cv::COLOR_RGB2BGR);
+    // }
+
+    // std::cout << "Frame size: " << frame.size() << std::endl;
+    // std::cout << "Overlay size: " << overlay.size() << std::endl;
+
+    for(const auto& detection : detections) {
+        int box_width = (detection.box.x_max - detection.box.x_min) * width + 1;
+        int box_height = (detection.box.y_max - detection.box.y_min) * height + 1;
+
+        if (box_width <= width && box_height <= height ) {
+            for (size_t i = 0; i < static_cast<size_t>(box_height); ++i) {
+                for (size_t j = 0; j < static_cast<size_t>(box_width); ++j) {
+                    auto cropped_mask_idx = static_cast<size_t>(i) * box_width + j;
+                    if (detection.mask[cropped_mask_idx]) {
+                        int overlayX = static_cast<int>(j) + static_cast<int>(detection.box.x_min * width);
```

Comment:
Variables should only be with lower case and with underscore - overlayX should be overlay_x etc.